src/service/world-api-adapter.js

```diff
@@ -245,11 +244,37 @@ module.exports = function (config) {
         },
 
         /**
+        * Get's the current (latest) world for the given user at the given group
+        *
+        */
+        getCurrentWorldForUser: function (userId, groupName) {
+            var dtd = $.Deferred();
+
+            this.getWorldsForUser(userId, { group: groupName })
+                .then(function (worlds) {
+                    // assume the most recent world as the 'active' world
+                    worlds.sort(function (a, b) { return new Date(b.lastModified) - new Date(a.lastModified); });
+                    var currentWorld = worlds[0];
+                    dtd.resolve(currentWorld, this);
+                })
+                .fail(dtd.reject);
+
+            return dtd.promise();
+        },
+
+        /**
         * Delete's the current run from the world
         *
         */
-        deleteRun: function () {
+        deleteRun: function (worldId, options) {
             throw new Error('not implemented');
+        },
+
+        newRunForWorld: function (worldId, options) {
+            return this.deleteRun(worldId)
```

Comment:
That's right, I just wanted to have logic code in place and then just implement the `deleteRun` so make it all work.